libraries/abstractions/secure_sockets/freertos_plus_tcp/iot_secure_sockets.c

```diff
@@ -496,8 +582,8 @@ Socket_t SOCKETS_Socket( int32_t lDomain,
 
     /* Ensure that only supported values are supplied. */
     configASSERT( lDomain == SOCKETS_AF_INET );
-    configASSERT( lType == SOCKETS_SOCK_STREAM );
-    configASSERT( lProtocol == SOCKETS_IPPROTO_TCP );
+    configASSERT( ( lType == SOCKETS_SOCK_STREAM ) || ( lType == SOCKETS_SOCK_DGRAM ) );
+    configASSERT( ( lProtocol == SOCKETS_IPPROTO_TCP ) || ( lProtocol == SOCKETS_IPPROTO_UDP ) );
```

Comment:
with the combination of values for ```configPLATFORM_SOCKET_UDP_SUPPORT ```, are these checks correct?

Reply:
Good Point. Here if application developer uses UDP flags by mistake, the check will pass. But other udp functions like sendTo and recvFrom will not compile.